src/hooks/useNavigationSearchBarCancelPress.ts

```diff
@@ -0,0 +1,36 @@
+import { useLayoutEffect } from 'react'
+import { Navigation, SearchBarCancelPressedEvent } from 'react-native-navigation'
+
+/**
+ * Registers listener to be called when the cancel button on the SearchBar from NavigationBar gets pressed.
+ * [more info](https://wix.github.io/react-native-navigation/api/events/#searchbarcancelpressed-ios-11-only)
+ */
+function useNavigationSearchBarCancelPress(
+  /**
+   * Function called each time the event is triggered.
+   */
+  handler: (event: SearchBarCancelPressedEvent) => void,
+
+  /**
+   * Component reference id. If provided it listens for event only for this screen.
+   */
+  componentId?: string
+) {
+  useLayoutEffect(() => {
+    const subscription = Navigation.events().registerSearchBarCancelPressedListener(
+      (event: SearchBarCancelPressedEvent) => {
+        const equalComponentId = event.componentId === componentId
+
+        if (componentId && !equalComponentId) {
+          return
+        }
+
+        handler(event)
```

Comment:
I noticed these lines are repeat multiple times. If all events extends from a common `Event` type who includes `componentId` then we could create a helper
```ts
function callWhenMatch(
  handler: (T extends Event) => void, 
  event: T, 
  componentId: string
): void {
  const equalComponentId = event.componentId === componentId

  if (componentId && !equalComponentId) {
    return
  }

  handler(event)
}

const subscription = Navigation.events().registerSearchBarCancelPressedListener(
  (event: SearchBarCancelPressedEvent) => {
    callWhenMatch(handler, event, componentId)
  }
)
```

Disclaimer: not sure if it's possible 😄

Reply:
Then maybe we can test `callWhenMatch` once and we can reduce the test cases for the rest of the _helpers_

Reply:
**Definitely Yes!** I was thinking on performing something similar when I tackle the componentId enforcement and the optional parameters we were talking about 😉